### PR TITLE
BAEL-6023: Throwing an Exception If an Optional Is Present

### PR DESCRIPTION
**Pull request by azhwani**
Simple PR

### CODE REVIEW

File: core-java-modules/core-java-optional/src/test/java/com/baeldung/optionaluses/UsesForOptionalUnitTest.java [hunk 1]
```diff
@@ -42,23 +43,49 @@ public void givenExistentUserId_whenFoundUserWithNameStartingWithMInRepositoryUs
...
+    void givenNonExistentUserId_whenSearchForUser_thenDoNotThrowException() {
+
+        final UserRepositoryWithOptional userRepositoryWithOptional = new UserRepositoryWithOptional();
+        String nonExistentUserId = "8";
+
+        assertDoesNotThrow(() -> {
+            userRepositoryWithOptional.throwExceptionWhenUserIsPresent(nonExistentUserId);
+        });
+
+    }
```
Comment (JavierPeris) [inline core-java-modules/core-java-optional/src/test/java/com/baeldung/optionaluses/UsesForOptionalUnitTest.java:89]: You can remove parentheses to convert from lambda statement to expression lambda:

`        assertDoesNotThrow(() -> userRepositoryWithOptional.throwExceptionWhenUserIsPresent(nonExistentUserId));`

Comment (azhwani) [reply core-java-modules/core-java-optional/src/test/java/com/baeldung/optionaluses/UsesForOptionalUnitTest.java]: Done

**Review by JavierPeris — APPROVED** [review]

File: core-java-modules/core-java-optional/src/test/java/com/baeldung/optionaluses/UsesForOptionalUnitTest.java [hunk 2]
```diff
@@ -50,15 +51,38 @@ public void givenExistentUserId_whenFoundUserWithNameStartingWithMInRepositoryUs
...
+
+    @Test
+    void givenExistentUserId_whenSearchForUser_thenThrowException() {
+
+        final UserRepositoryWithOptional userRepositoryWithOptional = new UserRepositoryWithOptional();
+        String existentUserId = "2";
+
+        assertThrows(UserFoundException.class, () -> {
+            userRepositoryWithOptional.throwExceptionWhenUserIsPresent(existentUserId);
+        });
```
Comment (JavierPeris) [inline core-java-modules/core-java-optional/src/test/java/com/baeldung/optionaluses/UsesForOptionalUnitTest.java:74]: I didn't wanted to be redundant but I was talking about all the lambda statement.

Comment (azhwani) [reply core-java-modules/core-java-optional/src/test/java/com/baeldung/optionaluses/UsesForOptionalUnitTest.java]: Oops, my bad. Done

**Pull request merged**